sdks/go/pkg/beam/transforms/sql/sql.go

```diff
@@ -79,6 +82,33 @@ func ExpansionAddr(addr string) Option {
 	}
 }
 
+const serviceGradleTarget = ":sdks:java:extensions:sql:expansion-service:shadowJar"
+
+// StartAutomatedExpansionService pulls a JAR from Maven containing
+// a Beam expansion service that handles the SQL transform URN,
+// creates a runner type to manage the service, starts the service,
+// and arranges the endpoint with the appropriate namespace tag.
+//
+// Returns the StopService() function for the runner (should be
+// deferred so the service is stopped before the program exits,)
+// the tagged expansion address, and an error.
+func StartAutomatedExpansionService() (func() error, string, error) {
```

Comment:
I think my question is why is this all done at the wrapper package level, rather than hidden away deeper? As implemented, every Jar based Xlang wrapper has to duplicate all this code and the code at the transform call itself, which is not a great experience, especially since there's nothing specific about it.

Ideally, wrapper authors should only be writing

defaultExpansion := xlangx.AutoNamespace + xlangx.Separator + <gradle target>

Or similar (or a wrapper function like Require does).

Then they pass that in, and the auto startup code will take care of the rest. Kinda like how the Python versions just take in the gradle target, and then the code takes care of the rest. In this case, everything should be happening within the scope of the auto HandlerFunc.

We do not need to optimize for repeated calls to the same expansion service. Users with complex enough pipelines hitting the same service should likely be standing up their own anyway.

If we don't want to use up "auto" just for jar startup we can always have something else to specify what kind of auto thing it's starting up: eg. "auto:jar:" to have it select sub name space for handling, or "autojar:" and just lean on the existing selector.

Reply:
Made a run at moving the logic deeper. Quick change since the logic itself was already written and working. We can change the namespace if you prefer but one major benefit to moving everything deeper is that we no longer have to export the namespace. Adding support should now be as easy as providing a gradle target for the expansion service a transform needs and calling `xlangx.UseAutomatedExpansionService()`